File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int[] map = new int[26];
        char[] input = s.toCharArray();
        for(char c : input){
            map[c-97]++;
        }
        
        for(int i = 0; i < s.length(); i++){
            if(map[input[i]-97] == 1)
                return i;
        }
        
        return -1;
    }
}

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.cs
public class Solution {
    public int FirstUniqChar(string s) {
        int[] map = new int[26];
        foreach(char c in s){
            map[c-97]++;
        }
        
        for(int i = 0; i < s.Length; i++){
            if(map[s[i]-97] == 1)
                return i;
        }
        
        return -1;
    }
}